test/datasets_utils.py

```diff
@@ -308,7 +319,7 @@ def create_dataset(
             info (Dict[str, Any]): Additional information about the injected fake data. See :meth:`.inject_fake_data`
                 for details.
         """
-        default_config = self._DEFAULT_CONFIG.copy()
+        default_config = self.DEFAULT_CONFIG.copy()
         if config is not None:
             default_config.update(config)
         config = default_config
```

Comment:
The meaning of `DEFAULT_CONFIG` seems to be "if a parameter isn't overriden in `config`, use the value specified in `DEFAULT_CONFIG` instead"

Where `DEFAULT_CONFIG` isn't just what we specified as a class attribute, because it's been updated in `SetUpClass()`.

So semantically, `DEFAULT_CONFIG` is very different from the entries in the `CONFIGS` list. It might be worth renaming `DEFAULT_CONFIG` to something more accurate?